packages/paste-core/components/chat-log/stories/index.stories.tsx

```diff
@@ -186,3 +199,43 @@ export const CustomizedMessages: StoryFn = () => (
     </Box>
   </CustomizationProvider>
 );
+
+// i want to give this story a state so you can see it flipping from spinner to file icon
+export const InboundChatMessageWithAttachment: StoryFn = () => (
+  <ChatMessage variant="inbound">
+    <ChatBubble>
+      <ChatAttachment attachmentIcon={DownloadIcon}>
```

Comment:
The way they can pass down overrides to the default behavior is to pass the icon as a function, like this:

```
<ChatAttachment attachmentIcon={() => <Spinner decorative={false} title="loading..." />}
```

That isn't immediately clear though, so if we go this route, we definitely need to talk about this in the documentation.

Reply:
Hmm.. maybe we should look into how the Input does it for insertBefore and insertAfter. The docs site has a correct example: https://paste.twilio.design/components/input#input-with-add-ons-prefix-suffix-text-or-icons

Reply:
This was a great idea! I've reworked it so that it uses insertBefore as its model. Consumers can now pass in components the way you initially suggested (the stories were changed to reflect this as well)